fed/proxy/grpc_proxy.py

```diff
@@ -0,0 +1,311 @@
+import asyncio
+import copy
+import cloudpickle
+import grpc
+import logging
+import threading
+import json
+import importlib.metadata
+from typing import Dict
+
+
+import fed.utils as fed_utils
+
+from fed.config import CrossSiloCommConfig, CrossSiloGrpcCommConfig
+from fed._private.compatible_utils import _compare_version_strings
+from fed._private.grpc_options import _DEFAULT_GRPC_CHANNEL_OPTIONS, _GRPC_SERVICE
+from fed.proxy.barriers import (
+    add_two_dim_dict,
+    get_from_two_dim_dict,
+    pop_from_two_dim_dict,
+    key_exists_in_two_dim_dict,
+    SendProxy,
+    RecvProxy
+)
+if _compare_version_strings(importlib.metadata.version('protobuf'), '4.0.0'):
```

Comment:
According to the comment https://github.com/ray-project/rayfed/pull/152#discussion_r1263227501 , maybe it's not a good idea to use importlib.metadata?

Reply:
Moved from "barriers.py" when resolving conflicts, I can resolve this when #155 is merged